java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/ByteFunctionHelpers.java

```diff
@@ -295,25 +295,25 @@ private static int memcmp(
   }
 
   /** Compute hashCode with the given {@link ArrowBuf} and start/end index. */
-  public static int hash(final ArrowBuf buf, long start, long end) {
+  public static long hash(final ArrowBuf buf, long start, long end) {
```

Comment:
Hash code does not strictly need changing and given the number of changes I'd rather skip it

Reply:
Sure I will revert that.